### PR TITLE
[FIX] crm : fix PLS settings and flow

### PR DESCRIPTION
**Pull request by dbeguin**
This commit applies multiples things:
- Fix PLS settings that were not loaded correctly (was not using technical field names)
- Add data for PLS settings to check all optional fields and set the start date.
- Set date pls field required
- use null value instead of empty for phone and email state fields to lessen the DB storage
- Remove Flanker that crashes anyway while validating a simple email address.
- Handle stage_id = False
- Remove default value for lead probability -> To let Bayes makes the magic.
- Set default pls date to today - 8 days to include all demo data.
- Set won should give 100% predicted probability

Task ID:2044539

### CODE REVIEW

File: addons/crm/models/res_config_settings.py [hunk 1]
```diff
@@ -48,7 +49,7 @@ def _inverse_pls_fields_str(self):
             we store the fields with a comma separated string into a Char config field """
         for setting in self:
             if setting.predictive_lead_scoring_fields:
-                setting.predictive_lead_scoring_fields_str = ','.join(setting.predictive_lead_scoring_fields.mapped('name'))
+                setting.predictive_lead_scoring_fields_str = ','.join(setting.predictive_lead_scoring_fields.mapped('field_id').mapped('name'))
```
Comment (dbeguin) [inline addons/crm/models/res_config_settings.py]: mapped(field_id.name')

Comment (tde-banana-odoo) [reply addons/crm/models/res_config_settings.py]: +1 mapped('field_id.name')

File: addons/crm/models/crm_lead.py [hunk 2]
```diff
@@ -227,11 +225,11 @@ def _compute_phone_state(self):
     @api.depends('email_from')
     def _compute_email_state(self):
         for lead in self:
-            email_state = 'empty'
+            email_state = False
             if lead.email_from:
                 email_state = 'incorrect'
                 for email in email_split(lead.email_from):
-                    if tools.email_validate(email):
+                    if tools.email_normalize(email):
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py]: Ouiiiiiiiiii

**Comment by dbeguin** [comment]
As that needs to be merged in saas-12.5 urgently, https://github.com/odoo/odoo/pull/37413 have been created.
Both should be merged at same time.

**Comment by lathama** [comment]
@dbeguin is this PR still needed?

**Comment by tde-banana-odoo** [comment]
Was done at https://github.com/odoo/odoo/commit/458355405fbc34c7d508b2a9d0407c2e9d939d87

**Pull request closed without merging**